### PR TITLE
[TravisCI] Add osx job

### PR DESCRIPTION
**Pull request by fghoussen**
## Pull request purpose

fixing issue #233 ?...
@cesss : you tell me what to do next !... not mac-based

### CODE REVIEW

**Comment by fghoussen** [comment]
same problem than with windows build !... tried to brew some package around ?... let's see

**Comment by fghoussen** [comment]
@cesss : what do you say ?

**Comment by cesss** [comment]
> @cesss : what do you say ?

Unfortunately I have no experience at automating builds and checks with travis --that's still in my TODO list 😄

On my Mac, it's what I said at the issue: this change in parpack.hpp does make the ICB build complete successfully, but then the two C++ tests fail (well, one of them fails, and the other gets stuck in an endless loop)

**Comment by fghoussen** [comment]
> I have no experience at automating builds and checks with travis

@cesss : Look at commits: now you have it. Not much about that anyway, it's about that : how to you brew you mac to get a clean env where to build arpack ? FC compiler ? packages ? brew options / conflicts ? ... I never had a mac so this is difficult for me to know

**Comment by sylvestre** [comment]
Maybe look at similar projects, how they are doing it!

**Comment by fghoussen** [comment]
> Maybe look at similar projects, how they are doing it!

Not clear / easy : step-by-step blind pushing...

**Comment by fghoussen** [comment]
@cesss : AFK, do you reproduce crashes when using `-D EXAMPLE=ON` ? (`[cz]ndrv*` are full fortran AFAIK, _not_ C++ related) I suspect compilers getting in each other's way : I ask for `llvm` but get `gcc`, how to make `[cf]lang` as default compilers on osx ?

**Comment by fghoussen** [comment]
Splitting this in 2 PRs : code fix + CI job

**Comment by fghoussen** [comment]
@cesss : do you reproduce the crash for the full-fortran `cnbdr3_ex` test ? can you get a stack ? 
seems impossible to get a stack from travis-ci box : sounds like the problem could be in lapack...

**Comment by fghoussen** [comment]
OK, does not work: stopping here. If somebody has an idea or can get a stack on osx, speak up !...

**Comment by fghoussen** [comment]
@sylvestre : 

1. d5ff2d7f99 fix build on osx when using autotool

2. if you accept (?) full build but only partial testing, you may get a green light on osx... not "great" but "better than nothing" (personal opinion) : you can at least test build is OK (which is not without d5ff2d7f99 for autotool) and at least a few tests

@cesss : can you get a stack on osx for failed tests like the full-fortran `cnbdr3_ex` for example ?

File: .travis.yml [hunk 1]
```diff
@@ -39,6 +40,38 @@ stages:
...
+    osx_image: xcode10.1
+    before_install:
+    - brew list -1 | while read line; do brew unlink $line; done;
+    - brew list gcc      || brew install gcc
+    - brew list cmake    || brew install cmake
+    - brew list openblas || brew install openblas
+    - brew list lapack   || brew install lapack
+    - brew list openmpi  || brew install openmpi
+    - brew list -1 | while read line; do brew link --overwrite $line; done;
+    # full build but only partial testing as some tests crash on osx (no stack: if you get one, open an issue https://github.com/opencollab/arpack-ng/issues)
```
Comment (fghoussen) [inline .travis.yml]: @sylvestre : caution, only partial testing

File: .travis.yml [hunk 2]
```diff
@@ -39,6 +40,38 @@ stages:
...
+    before_install:
+    - brew list -1 | while read line; do brew unlink $line; done;
+    - brew list gcc      || brew install gcc
+    - brew list autoconf || brew install autoconf
+    - brew list automake || brew install automake
+    - brew list openblas || brew install openblas
+    - brew list lapack   || brew install lapack
+    - brew list openmpi  || brew install openmpi
+    - brew list -1 | while read line; do brew link --overwrite $line; done;
+    # full build but only partial testing as some tests crash on osx (no stack: if you get one, open an issue https://github.com/opencollab/arpack-ng/issues)
```
Comment (fghoussen) [inline .travis.yml]: @sylvestre : caution, only partial testing here too

**Pull request merged**